src/widgets/modals/CashTransactionModal.js

```diff
@@ -154,80 +180,85 @@ export const CashTransactionModal = ({ onConfirm }) => {
+          </View>
+        </TouchableOpacity>
+        <TouchableOpacity style={localStyles.containerStyle} onPress={onPressAmount}>
+          <View style={localStyles.textContainerStyle}>
+            <Text style={localStyles.textStyle}>{amountText}</Text>
+          </View>
+          <View style={localStyles.iconContainerStyle}>
+            <PencilIcon />
+          </View>
+        </TouchableOpacity>
+        <PressReason />
+        <TouchableOpacity style={localStyles.containerStyle} onPress={onPressDescription}>
+          <View style={localStyles.textContainerStyle}>
+            <Text style={localStyles.textStyle}>{descriptionText}</Text>
+          </View>
+          <View style={localStyles.iconContainerStyle}>
+            <PencilIcon />
+          </View>
+        </TouchableOpacity>
+        <BottomCurrencyEditor
+          isOpen={isAmountModalOpen}
+          buttonText={buttonStrings.confirm}
+          value={amountBuffer}
+          placeholder={dispensingStrings.enter_the_amount}
+          onChangeText={onChangeAmount}
+          onConfirm={onSubmitAmount}
+        />
+        <BottomModalContainer
+          isOpen={isReasonModalOpen}
+          modalStyle={localStyles.bottomModalContainerStyle}
+        >
+          <GenericChoiceList
+            data={reasons}
+            keyToDisplay={CASH_TRANSACTION_KEYS.REASON}
+            onPress={onSubmitReason}
+            highlightValue={reason?.title}
+          />
+        </BottomModalContainer>
+        <BottomTextEditor
+          isOpen={isDescriptionModalOpen}
+          buttonText={buttonStrings.confirm}
+          value={descriptionBuffer}
+          placeholder={dispensingStrings.enter_a_description}
+          onChangeText={onChangeText}
+          onConfirm={onSubmitDescription}
+        />
+        <PageButton
+          text={buttonStrings.confirm}
+          onPress={onCreate}
+          isDisabled={!isValidTransaction}
+          disabledColor={WARM_GREY}
+          style={localStyles.okButton}
+          textStyle={localStyles.pageButtonTextStyle}
         />
-      </BottomModalContainer>
-      <BottomCurrencyEditor
-        isOpen={isAmountModalOpen}
-        buttonText={buttonStrings.confirm}
-        value={amountBuffer}
-        placeholder={dispensingStrings.enter_the_amount}
-        onChangeText={onChangeAmount}
-        onConfirm={onSubmitAmount}
-      />
-      <BottomModalContainer
-        isOpen={isReasonModalOpen}
-        modalStyle={localStyles.bottomModalContainerStyle}
+      </View>
+      <ModalContainer
+        title={dispensingStrings.choose_a_name}
+        isVisible={isNameModalOpen}
+        onClose={onCloseName}
       >
-        <GenericChoiceList
-          data={reasons}
-          keyToDisplay={CASH_TRANSACTION_KEYS.REASON}
-          onPress={onSubmitReason}
-          highlightValue={reason?.title}
+        <AutocompleteSelector
+          options={names}
+          queryString="name CONTAINS[c] $0"
```

Comment:
would possibly want to make it `firstName contains[c]... OR name... OR lastName....` -- alternatively we could make the `name` for a patient equal to `lastName, firstName` rather than `""` as it is now

Reply:
e7996746d24ab5f37b3199defc234850c436e4cf.

Reply:
I forgot about this - in desktop `[name]name` is already set to this for patients. The `createRecord` case has also now been updated https://github.com/openmsupply/mobile/blob/5c86d528ff49ef4cef189881277181ff0fa29236/src/database/utilities/createRecord.js#L123-L136

so this isn't needed

Reply:
Can we get these changes into `#feature-dispensary` (alternatively can we get this approved and resolve at merge)?

Reply:
the link above is from `#feature-dispensary` ? confused sorry

Reply:
It's been merged now 👍 .